framework/db/ActiveQuery.php

```diff
@@ -225,6 +225,10 @@ public function populate($rows)
             }
         }
 
+        if ($this->collectionClass !== null) {
+            return new $this->collectionClass($models);
```

Comment:
I'd suggest to move this line into a separate protected method. Then it could be overridden with custom logic.
Use case: I want to pass original `Query` object in my collection as a 2nd argument of the constructor.